### PR TITLE
Issue/166 paginate data table document

### LINKED ISSUE #166: Paginate data to view on table document
Paginate data to view on document list using query cursors to limit length of data to be recieved

Reference:
https://firebase.google.com/docs/firestore/query-data/query-cursors

### PR DESCRIPTION
**Pull request by detautama**
Fixes #166 

FIX pagination document

### CODE REVIEW

File: angular/projects/admin/src/app/admin/document/document-list/document-list.component.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ import { tap } from 'rxjs/operators';
 })
 export class DocumentListComponent implements OnInit, OnDestroy, AfterViewInit {
   total_count: number;
+  lastPageIndex: Number = 0;
```
Comment (DennisAlund) [inline angular/projects/admin/src/app/admin/document/document-list/document-list.component.ts]: Use the type `number` with all small letters (https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html).

image

File: angular/projects/admin/src/app/admin/document/document-list/document-list.component.ts [hunk 2]
```diff
@@ -62,8 +68,8 @@ export class DocumentListComponent implements OnInit, OnDestroy, AfterViewInit {
     const url = `/_/admin/document/${documentId}`;
     this.router.navigateByUrl(url);
   }
-  setTotalCount (dataSource: Object) {
-    const refNumEntries = dataSource['documentType'].documentCount;
+  setTotalCount(dataSource: Object) {
+    const refNumEntries = dataSource['documentType'].numEntries;
```
Comment (DennisAlund) [inline angular/projects/admin/src/app/admin/document/document-list/document-list.component.ts]: Seems like the previous use of `documentCount` is should be used. Perhaps it's just the database that's having the wrong data.

File: angular/projects/admin/src/app/services/document.service.ts [hunk 3]
```diff
@@ -111,15 +123,36 @@ export class DocumentService {
         query = query.orderBy(queryOpts.orderBy.field, queryOpts.orderBy.sortOrder);
       }
 
+      if (!nextPage && this.firstDoc[this.firstDoc.length - 2]) { // back
```
Comment (DennisAlund) [inline angular/projects/admin/src/app/services/document.service.ts]: What is the `-2` value representing?

File: angular/projects/admin/src/app/services/document.service.ts [hunk 4]
```diff
@@ -98,9 +100,19 @@ export class DocumentService {
...
+    documentTypeId: string, status: string, queryOpts: DocumentTypeQueryOptions = {}, nextPage?: Boolean, pageIndex?: number
+  ): Observable<Document[]> {
     console.log(`[DocumentService:getDocumentTypeFields] ${documentTypeId}, query=${JSON.stringify(queryOpts)}`);
 
+    console.log('[Next page] ', nextPage);
+    console.log('[page index] ', pageIndex);
+
+    if (pageIndex === 0) {
+      this.firstDoc = [];
+      this.lastDoc = '';
```
Comment (DennisAlund) [inline angular/projects/admin/src/app/services/document.service.ts]: Is `firstDoc` and `lastDoc` representing different things? Seems strange that name is very similar but one is an array and the other a string.

File: angular/projects/admin/src/app/admin/document/document-list/document-list.component.ts (hunk 2 above)
Comment (detautama) [reply angular/projects/admin/src/app/admin/document/document-list/document-list.component.ts]: Resolve on this commit https://github.com/oddbit/tanam/pull/193/commits/50bb2072b77606029542acdf6e0c385b91328c33

File: angular/projects/admin/src/app/admin/document/document-list/document-list.component.ts (hunk 1 above)
Comment (detautama) [reply angular/projects/admin/src/app/admin/document/document-list/document-list.component.ts]: Resolve on this commit https://github.com/oddbit/tanam/pull/193/commits/e4fd812a1d7a0d05d2b802e879fae0544502fb07

**Pull request closed without merging**